### ISSUE TITLE
Access the already existing Windows clipboard

### ISSUE THREAD

**Issue by Hofei90** [spyoungtech/pyperclip3#7]
Can I also access the already existing Windows clipboard when starting the script.
Currently I have to copy something to the clipboard with pyclip first, otherwise I get the following error message

```
Traceback (most recent call last):
  File "C:/Users/***/PycharmProjects/test/clippy.py", line 4, in <module>
    cb_data = pyclip.detect_clipboard().paste()
  File "C:\Users\***\AppData\Local\Programs\Python\Python38-32\lib\site-packages\pyclip\win_clip.py", line 219, in paste
    d = clip.GetClipboardData(format)
pywintypes.error: (1418, 'GetClipboardData', 'Der Thread hat keine Zwischenablage geöffnet.')
```

The slightly modified sample code

import pyclip

```
#pyclip.copy("hello clipboard") # copy data to the clipboard
cb_data = pyclip.detect_clipboard().paste()
print(cb_data)  # b'hello clipboard'
cb_text = pyclip.paste(text=True)  # paste as text
print(cb_text)  # 'hello clipboard'

#pyclip.clear() # clears the clipboard contents
#assert not pyclip.paste()
```

**Comment by spyoungtech** [comment]
Interesting. I am able to reproduce, but only in certain circumstances. 🧐 

For instance if I copy from notepad.exe or the URL from a browser, this works fine.

But if I copy from my IDE or webpages, I get this error.

Looking into this...

**Comment by spyoungtech** [comment]
Ok. I spotted the error.

Applications can provide many clipboard formats onto the clipboard, even formats that they make up. `pyclip` (and any application for that matter) can only reasonably handle formats it knows about, I.E. standard clipboard formats or formats it creates or is otherwise aware of.

Pyclip tries to handle this properly by enumerating all possible formats and requesting the best format it knows how to handle (this is how applications are generally expected to work with the Windows clipboard)

Currently there is a bug in pyclip where it does not correctly handle the `CF_LOCALE` format (and possibly other formats it does not properly handle) that it allows to be passed to `GetClipboardData`.

I'm still not quite sure why _this particular error_ is thrown, but I think I'll be able to work around it in any case.

**Comment by spyoungtech** [comment]
There is a bug in pyclip where it might accidentally close the clipboard before it attempts to retrieve the data.

`paste` [uses a helper function](https://github.com/spyoungtech/pyclip/blob/main/pyclip/win_clip.py#L208) to enumerate the clipboard format.

However, that helper function causes the clipboard to be closed when it returns, causing the error when it subsequently tries to retrieve that data.

Additionally, PyClip will try to attempt to handle some formats incorrectly, it seems, though this is a different problem.

I should be able to provide a patch for this soon.

**Comment by spyoungtech** [comment]
Should be fixed with #8 and has been released with latest version: 0.5.3

@Hofei90 let me know if that fixes it for you.

**Comment by Hofei90** [comment]
Yes, it works for me.
Thanks for the quick correction - Great Job.
